docs/python/tutorial-fastapi.md

```diff
@@ -0,0 +1,595 @@
+
+If you are already working on **GitHub Codespaces** with our existing template, you can skip directly to the ["Replace the database"](#replace-the-database) section.
+
+If you are on Windows, you can work with Redis by setting up either a [Docker container](https://www.docker.com/products/docker-desktop/), a [GitHub Codespace](https://github.com/features/codespaces) or through [WSL](https://learn.microsoft.com/en-us/windows/wsl/) (Windows Subsystem for Linux). In this tutorial we will cover the set up using Docker containers, but you can refer to the [section above](#github-codespaces) for instructions on how to set up a GitHub Codespace.
+
+Otherwise, if you are on a Linux or a macOS machine, you can install Redis by following the [instructions on their website](https://redis.io/docs/getting-started/), and then skip to the ["Replace the database"](#replace-the-database) section.
+
+### Setting up a Docker Container on Windows
+The Dev Container extension for VS Code offers a simple way to configure a Linux-based workspace in your Windows machine, putting your entire project and its dependencies as well as tools you need for your project into a tidy container.
+For the steps below, make sure you have the following requirements installed on your machine:
+
+#### Requirements
+-	[Docker for Windows](https://www.docker.com/)
+-	[Dev Containers extension for VS Code](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers)
+
+#### Create the Dev Container configuration
+1. Open the Command Palette and run the “Dev Container: Add Dev Container Configuration Files…"
+2. Select "Python 3"
+    ![Python 3 option selected in the Dev Container configuration files list.](images/fastapi-tutorial/devcontainers_python3.png)
+3. Select the default version.
+
+We can optionally install features to be included in the container. For this tutorial, we will install [Redis Server](https://github.com/itsmechlark/features/tree/main/src/redis-server), which is a community contributed feature that installs and adds the proper dev container set up for Redis.
+
+4. Select "Redis Server" as an additional feature to be installed, press "OK", and then select "Keep Defaults".
+    ![Redis Server option selected in the Dev Container configuration files list.](images/fastapi-tutorial/devcontainers_redis_server_feature.png)
+
+This will create a `.devcontainer` folder in your workspace, with a `devcontainer.json` file. Let's make some edits to this file so the container setup can include steps such as installing the VS Code extensions we'll need as well as the project dependencies.
+
+5. Open the `devcontainer.json` file.
+
+6. Locate the content below and remove the comment (`//`) from that line, so the dependencies can be installed once the container is created:
```

Comment:
I also had to add a `,` after the features block to handle the red squiggles. Not sure if it is worth calling something out here so the user is able to easily handle this error.